minor: fix broken links

nick-mancuso opened the pull request:
Noticed at https://github.com/checkstyle/checkstyle/runs/6930711012?check_suite_focus=true

Two remaining "broken" links:

One (mojohaus) has been suppressed, since the plugin needs to update the link on their end.
I can't figure out what is going on with the other, I have no problem accessing the site:

```
➜  checkstyle git:(fix-broken-links) curl https://checkstyle.sourceforge.io/version/6.18/ -I 
HTTP/2 200 
date: Fri, 17 Jun 2022 15:43:47 GMT


```

Remaining "broken" links:
```
------------ grep of linkcheck.html--BEGIN
<td><i><a class="externalLink" href="https://checkstyle.sourceforge.io/version/6.18/">https://checkstyle.sourceforge.io/version/6.18/</a>: 403 Forbidden</i></td></tr></table></td></tr>
<td><i><a class="externalLink" href="http://www.mojohaus.org/versions-maven-plugin/">http://www.mojohaus.org/versions-maven-plugin/</a>: 301 Moved Permanently</i></td></tr></table></td></tr>
------------ grep of linkcheck.html--END
Exit code:2

```

Edit: failing sourceforge links have been removed

nick-mancuso commented:
Github, generate site

nick-mancuso commented on pom.xml:
```diff
@@ -2031,6 +2031,7 @@
             <excludedLink>https://paypal.com/</excludedLink>
             <!-- temporal suppress till plugin update link in his source -->
             <excludedLink>http://www.mojohaus.org/exec-maven-plugin</excludedLink>
+            <excludedLink>http://www.mojohaus.org/versions-maven-plugin</excludedLink>
```
Same as above, plugin needs to update their link.

nick-mancuso commented:
I am not sure what we want to do about this link: `<td><i><a class="externalLink" href="https://checkstyle.sourceforge.io/version/6.18/">https://checkstyle.sourceforge.io/version/6.18/</a>: 403 Forbidden</i></td></tr></table></td></tr>
`

rnveach commented:
@nick-mancuso 
> checkstyle.sourceforge.io

We should not be linking to our sourceforge site. We moved our site to https://checkstyle.org/ and everything should be linked to that.

romani commented on src/xdocs/index.xml.vm:
```diff
@@ -554,9 +554,9 @@
+              <td><a href="https://vim8.org/">Vim editor</a></td>
               <td>Xandy Johnson</td>
-              <td><a href="https://vim.sourceforge.io/scripts/script.php?script_id=448">
+              <td><a href="https://vim8.org/">
```
If plugin is gone.
We should remove completely.

This section about plugins and links to them.

nick-mancuso replied:
Done.

romani reviewed (changes requested):
Items

nick-mancuso commented:
> @nick-mancuso
> 
> > checkstyle.sourceforge.io
> 
> We should not be linking to our sourceforge site. We moved our site to https://checkstyle.org/ and everything should be linked to that.

Thanks for pointing this out. I have removed this entire section, since we have all release note documentation in one place now.

romani commented on src/xdocs/index.xml.vm:
```diff
@@ -553,13 +535,6 @@
-            <tr>
-              <td><a href="https://vim.sourceforge.io/">Vim editor</a></td>
-              <td>Xandy Johnson</td>
-              <td><a href="https://vim.sourceforge.io/scripts/script.php?script_id=448">
```
https://www.vim.org/scripts/script.php?script_id=448

romani replied:
First link at https://www.google.com/search?q=Vim+editor+checkstyle+plugin&oq=Vim+editor+checkstyle+plugin+&aqs=chrome..69i57j33i160l5j33i299.7770j0j7&client=ms-android-google&sourceid=chrome-mobile&ie=UTF-8

nick-mancuso replied:
I misunderstood above, I thought you checked and plugin was gone, so I did not look. Done.

romani reviewed (changes requested):
Items

romani commented on src/xdocs/index.xml.vm:
```diff
@@ -69,24 +69,6 @@
 
-    <section name="Previous Version Documentation">
-      <p>
-        Current site contains documentation only for latest, released version.
```
hmm, this section is good. It is required to let people know how to see documentation for old versions.

nick-mancuso replied:
It is now intuitive, i.e. keep scrolling to bottom of release information and see link to last interval, visit that link, repeat.

romani replied:
lets move discussion on this section out of this PR.
I do not understand your point, more details are required. I do not want to block this pr due to this section

nick-mancuso replied:
Section is restored, issue created at https://github.com/checkstyle/checkstyle/issues/11754

romani reviewed (changes requested):
items:
